torch/csrc/api/include/torch/data/datasets/tensor.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <torch/data/datasets/base.h>
+#include <torch/data/example.h>
+#include <torch/tensor.h>
+
+#include <cstddef>
+#include <vector>
+
+namespace torch {
+namespace data {
+namespace datasets {
+class TensorDataset : public Dataset<TensorDataset, TensorExample> {
+ public:
+  explicit TensorDataset(std::vector<Tensor> tensors)
```

Comment:
I think this was maybe meant to support multiple inputs / targets, and not only two? This was currently extended in the python implementation (accept a `*args` instead of a `input` / `target`)